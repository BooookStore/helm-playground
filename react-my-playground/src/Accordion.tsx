import { useState } from "react";
import accordionContents from "./accordionContents";

export default function Accordion() {
  const [coffeePanelShow, setCoffeePanelShow] = useState(true);

  return (
    <div className="flex gap-5">
      <Panel
        content={accordionContents[0]}
        show={coffeePanelShow}
        onToggleButtonClick={() => setCoffeePanelShow(!coffeePanelShow)}
      />
      <Panel
        content={accordionContents[1]}
        show={!coffeePanelShow}
        onToggleButtonClick={() => setCoffeePanelShow(!coffeePanelShow)}
      />
    </div>
  );
}

function Panel({
  content,
  show,
  onToggleButtonClick,
}: {
  content: {
    title: string;
    description: string;
  };
  show: boolean;
  onToggleButtonClick: () => void;
}) {
  const buttonText = show ? "hide" : "show";

  return (
    <section className="w-1/2">
      <h2>{content.title}</h2>
      <button
        className="bg-indigo-700 hover:bg-indigo-400 py-1 px-2 text-slate-50 rounded drop-shadow"
        onClick={onToggleButtonClick}
      >
        {buttonText}
      </button>
      {show && <p>{content.description}</p>}
    </section>
  );
}
