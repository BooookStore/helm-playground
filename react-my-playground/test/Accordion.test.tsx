import React from "react";
import { describe, test, afterEach, expect } from "vitest";
import { render, screen, cleanup, fireEvent } from "@testing-library/react";

import Accordion from "../src/Accordion";

describe("Accordion", () => {
  test("render", () => {
    const { container } = render(<Accordion />);

    // initial render
    expect(container).toMatchSnapshot();

    // toggle coffee section to tea section
    const button = screen.getByText("show");
    fireEvent.click(button);
    expect(container).toMatchSnapshot();
  });

  afterEach(cleanup);
});
