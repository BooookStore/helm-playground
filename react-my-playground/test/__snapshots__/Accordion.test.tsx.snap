// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Accordion > render 1`] = `
<div>
  <div
    class="flex gap-5"
  >
    <section
      class="w-1/2"
    >
      <h2>
        Coffee
      </h2>
      <button
        class="bg-indigo-700 hover:bg-indigo-400 py-1 px-2 text-slate-50 rounded drop-shadow"
      >
        hide
      </button>
      <p>
        コーヒーは、コーヒー豆から抽出される飲み物であり、世界中で広く愛されています。豆を挽いてお湯で淹れることで、香り高く濃厚な味わいが楽しめます。
      </p>
    </section>
    <section
      class="w-1/2"
    >
      <h2>
        Tea
      </h2>
      <button
        class="bg-indigo-700 hover:bg-indigo-400 py-1 px-2 text-slate-50 rounded drop-shadow"
      >
        show
      </button>
    </section>
  </div>
</div>
`;

exports[`Accordion > render 2`] = `
<div>
  <div
    class="flex gap-5"
  >
    <section
      class="w-1/2"
    >
      <h2>
        Coffee
      </h2>
      <button
        class="bg-indigo-700 hover:bg-indigo-400 py-1 px-2 text-slate-50 rounded drop-shadow"
      >
        show
      </button>
    </section>
    <section
      class="w-1/2"
    >
      <h2>
        Tea
      </h2>
      <button
        class="bg-indigo-700 hover:bg-indigo-400 py-1 px-2 text-slate-50 rounded drop-shadow"
      >
        hide
      </button>
      <p>
        紅茶は、茶葉から抽出される飲み物であり、世界中で人気があります。茶葉をお湯で浸し、適切な時間で淹れることで、豊かな風味とアロマが楽しめます。
      </p>
    </section>
  </div>
</div>
`;

exports[`Accordion > render 3`] = `
<div>
  <div
    class="flex gap-5"
  >
    <section
      class="w-1/2"
    >
      <h2>
        Coffee
      </h2>
      <button
        class="bg-indigo-700 hover:bg-indigo-400 py-1 px-2 text-slate-50 rounded drop-shadow"
      >
        hide
      </button>
      <p>
        コーヒーは、コーヒー豆から抽出される飲み物であり、世界中で広く愛されています。豆を挽いてお湯で淹れることで、香り高く濃厚な味わいが楽しめます。
      </p>
    </section>
    <section
      class="w-1/2"
    >
      <h2>
        Tea
      </h2>
      <button
        class="bg-indigo-700 hover:bg-indigo-400 py-1 px-2 text-slate-50 rounded drop-shadow"
      >
        show
      </button>
    </section>
  </div>
</div>
`;
